src/enhancedApp.ts

```diff
@@ -0,0 +1,87 @@
+import { RepoEntitlementsRepository } from './repositories/repoEntitlementsRepository';
+import { ISSOConnector, OktaConnector } from './services/sso';
+import c from 'config';
+import * as mongodb from 'mongodb';
+import { FileSystemServices } from './services/fileServices';
+import { JobValidator } from './job/jobValidator';
+import { RepoBranchesRepository } from './repositories/repoBranchesRepository';
+import { EnhancedJobManager } from './job/enhancedJobManager';
+
+let db: mongodb.Db;
+let consoleLogger: ConsoleLogger;
+let fileSystemServices: FileSystemServices;
+let jobCommandExecutor: JobSpecificCommandExecutor;
+let githubCommandExecutor: GithubCommandExecutor;
+let jobRepository: JobRepository;
+let hybridJobLogger: HybridJobLogger;
+let repoEntitlementRepository: RepoEntitlementsRepository;
+let jobValidator: JobValidator;
+let cdnConnector: K8SCDNConnector;
+let ssmConnector: ParameterStoreConnector;
+
+let repoConnector: GitHubConnector;
+let jobHandlerFactory: JobHandlerFactory;
+let jobManager: JobManager;
+let repoBranchesRepo: RepoBranchesRepository;
+let ssoConnector: ISSOConnector;
+
+async function init(): Promise<void> {
+  const atlasURL = `mongodb+srv://${c.get('dbUsername')}:${c.get('dbPassword')}@${c.get(
+    'dbHost'
+  )}/?retryWrites=true&w=majority`;
+  const client = new mongodb.MongoClient(atlasURL);
+  await client.connect();
+  db = client.db(c.get('dbName'));
+  consoleLogger = new ConsoleLogger();
+  fileSystemServices = new FileSystemServices();
+  jobCommandExecutor = new JobSpecificCommandExecutor();
+  githubCommandExecutor = new GithubCommandExecutor();
+  jobRepository = new JobRepository(db, c, consoleLogger);
+  hybridJobLogger = new HybridJobLogger(jobRepository);
+  ssmConnector = new ParameterStoreConnector();
+  repoEntitlementRepository = new RepoEntitlementsRepository(db, c, consoleLogger);
+  repoBranchesRepo = new RepoBranchesRepository(db, c, consoleLogger);
+  jobValidator = new JobValidator(fileSystemServices, repoEntitlementRepository, repoBranchesRepo);
+  ssoConnector = new OktaConnector(c, consoleLogger);
+  cdnConnector = new K8SCDNConnector(c, consoleLogger, ssmConnector, ssoConnector);
+  repoConnector = new GitHubConnector(githubCommandExecutor, c, fileSystemServices, hybridJobLogger);
+  jobHandlerFactory = new JobHandlerFactory();
+
+  const cleanUp = async () => {
+    await jobManager.stop();
+    if (client) {
+      await client.close();
+      consoleLogger.info('App', '\nServer has closed mongo client connection');
+    }
+    process.exit(0);
+  };
+
+  jobManager = new EnhancedJobManager(
+    c,
+    jobValidator,
+    jobHandlerFactory,
+    jobCommandExecutor,
+    jobRepository,
+    cdnConnector,
+    repoConnector,
+    fileSystemServices,
+    hybridJobLogger,
+    repoBranchesRepo,
+    cleanUp
+  );
+
+  try {
+    jobManager.start();
+  } catch (err) {
+    consoleLogger.error('App', `ERROR: ${err}`);
```

Comment:
Just wondering if we'd ever want to make this `enhancedApp` to differentiate between any caught errors here for the regular app.